src/components/PopupWithForm.js

```diff
@@ -20,6 +21,14 @@ export default class PopupWithForm extends Popup {
     super.close();
   }
 
+  setLoadingText(isLoading, defaultText = "Save") {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = defaultText;
+    }
+  }
+
```

Comment:
```js

      this.close();
      this._popupForm.reset();
```

Please, delete them from lines 37 and 38

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

You need to **reset** the form only  in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened, and the user can try to send the data again to the server without having to type the data into the inputs once more.